cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginProperties.java

```diff
@@ -33,19 +35,40 @@
 
   // Currently only support String->String map.
   private final Map<String, String> properties;
+  private final Set<String> macros;
+
+  private PluginProperties(Map<String, String> properties, @Nullable Set<String> macros) {
+    this.properties = properties;
+    this.macros = macros;
+  }
 
   public static Builder builder() {
     return new Builder();
   }
 
   private PluginProperties(Map<String, String> properties) {
-    this.properties = properties;
+    this(properties, null);
   }
 
   public Map<String, String> getProperties() {
     return properties;
   }
 
+  @Nullable
+  public Set<String> getMacros() {
```

Comment:
while we're doing this, we should make these objects rather than strings. I think it makes sense to keep track of macro functions and which field it came from instead of just the lookup macros. An object instead of a string also makes this easier to extend in the future.